src/resolver.ts

```diff
@@ -1006,6 +1009,21 @@ export class Resolver extends DiagnosticEmitter {
     ctxType: Type = Type.auto,
     /** How to proceed with eventual diagnostics. */
     reportMode: ReportMode = ReportMode.REPORT
+  ): Type | null {
+    if (this.resolvingExpressions.has(node)) return null;
+
+    this.resolvingExpressions.add(node);
+    const resolved = this.unsafeResolveExpression(node, ctxFlow, ctxType, reportMode);
+    this.resolvingExpressions.delete(node);
+    return resolved;
```

Comment:
Can cache the property access in a local here:
```suggestion
    const resolvingExpressions = this.resolvingExpressions;
    if (resolvingExpressions.has(node)) return null;
    resolvingExpressions.add(node);
    const resolved = this.unsafeResolveExpression(node, ctxFlow, ctxType, reportMode);
    resolvingExpressions.delete(node);
    return resolved;
```